File: 프로그래머스/unrated/181930. 주사위 게임 2/주사위 게임 2.java

import static java.lang.Math.*;

class Solution {
    public int solution(int a, int b, int c) {
        int answer = 0;
        
        int[] num = {a,b,c};
        
        int two_dup = 0;
        int three_dup = 0;

        for (int i : num) {
            answer += i;
        }

        if(num[0] == num[1] || num[1] == num[2] || num[0] == num[2]) {
            for(int i : num) {
                two_dup += (int) (pow(i,2));
            }
            answer *= two_dup;
            if(num[0] == num[1] && num[1] == num[2]) {
                for(int i : num) {
                    three_dup += (int) (pow(i,3));
                }
                answer *= three_dup;
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.py
# 주사위 3개
# 3개의 주사위를 굴렸을 때 값이 a, b, c 
# a b c 모두 값이 다름 -> a + b + c
# 두 값이 동일, 나머지 값 다름 -> (a + b + c) * (a**2 + b**2 + c**2)
# 모든 값 동일 -> (a + b + c) * (a**2 + b**2 + c**2) * (a**3 + b**3 + c**3)

def three(a, b, c):
    return two(a,b,c) * (a**3 + b**3 + c**3)

def two(a, b, c):
    return one(a,b,c) * (a**2 + b**2 + c**2)

def one(a, b, c):
    return a + b + c

def solution(a, b, c):
    answer = 0
    if a == b: # 2개 동일
        if b == c: # a == b == c
            answer = three(a,b,c)
        else: # a == b != c
            answer = two(a,b,c)
    else:
        if b == c: # a != b == c
            answer = two(a,b,c)     
        else: # a != b != c
            answer = two(a,b,c) if a == c else one(a,b,c)
    
    return answer